### ISSUE TITLE
Z-fighting bug

### ISSUE THREAD

**Issue by vvs-** [XProger/OpenLara#208]
There should be leaves on the wall in a big room with wolves and timed doors in _Caves_. Due to z-fighting there is a wall texture instead. This is almost the same as in corresponding OpenTomb bug (but it flickers there).

BTW in OT there used to be z-fighting problems in _Lost Valley_ too. I didn't check it yet because it's hard to fight all those baddies there (T-Rex). Are there any means that could help to make Lara invincible or eliminate all enemies altogether? OT has excellent debugging aids for such tasks of which OL would benefit too.

**Comment by XProger** [comment]
you can use O key for DOZY mode in OL

**Comment by vvs-** [comment]
Thanks for the suggestion. I think there is the same bug in _Lost Valley_ but to be sure I must complete those levels in original TR too which doesn't have such cheats.

Here is screenshot from original TR in _Caves_. To be fair there is z-fighting even there but it's not so obvious.
![tomb_001](https://user-images.githubusercontent.com/5171934/65449022-c4a70a00-de42-11e9-8e1d-bd4d8d922c89.png)

**Comment by vvs-** [comment]
Yes, there is the same z-fighting bug in _Lost Valley_ and there is no sign of it in the original engine.
![tomb_000](https://user-images.githubusercontent.com/5171934/65529284-84a55d00-defe-11e9-8fea-af3994be6829.png)

**Comment by XProger** [comment]
It's content problem, the original game doesn't use z-buffer, TombATI and Android/iOS ports has the same z-fighting bugs... so I don't know how I can fix it.

**Comment by Popov72** [comment]
Where is it exactly in the Lost Valley level? Can you give the camera coordinates?

Thanks.

**Comment by vvs-** [comment]
It's under the broken bridge to one of the cogs, near T-Rex.

**Comment by vvs-** [comment]
@XProger understood. But what if you give some textures more priority? That way wall and floor textures will only be rendered if there is no z-fighting there.

**Comment by XProger** [comment]
It'll be looking weird, because in bridge parts case they are really under ground level

**Comment by Popov72** [comment]
Using appropriate zbias should do it (PolygonOffset in openGL).

**Comment by vvs-** [comment]
Also, run-time patching could do wonders :smile:
